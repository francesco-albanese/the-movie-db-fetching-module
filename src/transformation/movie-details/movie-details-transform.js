import { isEmpty } from 'lodash-es'
import { getMovieImage } from '#utils'

export const transformMovieDetailsResponse = (data = {}) => {

  if (isEmpty(data) || data.status_code) {
    //an error occurred or the movieid is not valid
    return {}
  } 
  
  const {
    backdrop_path,
    genres,
    overview,
    poster_path,
    release_date,
    runtime,
    title,
    vote_average
  } = data

  const genreNames = genres.map(({ name }) => name)

  const background = isEmpty(backdrop_path)
    ? 'https://via.placeholder.com/1096x617.png?text=No+Image+Available'
    : getMovieImage(backdrop_path)
    
  const poster = isEmpty(poster_path) 
    ? 'https://via.placeholder.com/411x617.png?text=No+Image+Available'
    : getMovieImage(poster_path)

  return {
    background,
    description: overview,
    genres: genreNames,
    poster,
    release_date,
    rating: vote_average,
    runtime,
    title
  }
}