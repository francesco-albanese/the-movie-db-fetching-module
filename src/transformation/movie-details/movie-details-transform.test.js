import { transformMovieDetailsResponse } from './movie-details-transform'
import * as Fixtures from './__fixtures__/movie-details.transform.fixtures'

describe('transformMovieDetailsResponse', () => {
  it('returns an empty object if data is missing', () => {
    expect(transformMovieDetailsResponse()).toEqual({})
  })

  it('returns a transformed movie', () => {
    const result = transformMovieDetailsResponse(Fixtures.data)
    expect(result).toMatchSnapshot()
  })
})