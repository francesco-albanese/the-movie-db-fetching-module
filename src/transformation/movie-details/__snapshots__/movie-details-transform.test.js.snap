// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`transformMovieDetailsResponse returns a transformed movie 1`] = `
Object {
  "background": "undefined/cmAHUFBqg8VO0InX9QdDTZQSg9H.jpg",
  "description": "Siamo nel 1916. In un piccolo villaggio tedesco, il professore di liceo Kantorek esalta gli ideali bellici e patriottici e convince i propri allievi ad arruolarsi (tranne un giovane chiamato Giuseppe Behn, persuaso poi dai suoi compagni), ma il loro entusiasmo scema immediatamente all'eseguire, durante l'addestramento, gli ordini del sottufficiale Himmelstoss, che con i suoi modi ed il carattere austero educherà i ragazzi alla vera realtà del fronte, ben diversa dalla retorica del professore.  I soldati vengono poi assegnati ad un'unità di veterani, che nei loro confronti si sentono superiori e sono meno accomodanti; questi, senza cibo da due giorni, ridacchiano con un po' di malizia quando i \\"pulcini\\" chiedono loro qualcosa da mangiare. Fortunatamente, hanno dalla loro parte un soldato particolarmente abile nel trovare cibarie, chiamato Stanislaw Katczinsky.",
  "genres": Array [
    "Dramma",
    "Guerra",
  ],
  "poster": "undefined/gmNUUGxidAUNDBH4BUvDo1JFNyX.jpg",
  "rating": 7.7,
  "release_date": "1930-04-29",
  "runtime": 133,
  "title": "All'ovest niente di nuovo",
}
`;
